.github/workflows/build_and_test.yaml

```diff
@@ -0,0 +1,180 @@
+      GYP_MSVS_VERSION: "2019"
+      GYP_MSVS_OVERRIDE_PATH: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise"
+
+    steps:
+      - name: Compute ref
+        # We could be building from a workflow dispatch (manual run or triggered
+        # by another workflow), or from a pull request.  Subsequent steps can
+        # refer to $TARGET_REF to determine the correct ref in all cases.
+        run: |
+          echo "TARGET_REF=${{ github.event.inputs.ref || github.ref }}" >> $GITHUB_ENV
+
+      - name: Configure git to preserve line endings
+        # Fix issues with CRLF in test outputs on Windows by explicitly setting
+        # this core.autocrlf config.  Without this, tests will later fail
+        # because the "golden" test outputs in the source code will not have
+        # the correct line endings.
+        run: |
+          git config --global core.autocrlf false
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          path: src
+          ref: ${{ env.TARGET_REF }}
+
+      - name: Install depot tools
+        # NOTE: can't use actions/checkout here because this is not hosted on
+        # GitHub.
+        run: |
+          git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
+          echo "${GITHUB_WORKSPACE}/depot_tools" >> $GITHUB_PATH
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 2.7
+
+      - name: Install macOS SDK 10.3 (macOS only)
+        if: matrix.os_name == 'osx'
+        # NOTE: macOS 11 doesn't work with our (old) version of Chromium build,
+        # and the latest Chromium build doesn't work with Packager's build
+        # system.  To work around this, we need an older SDK version, and to
+        # get that, we need an older XCode version.  XCode 10.3 has SDK 10.14,
+        # which works.
+        uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: 10.3
+
+      - name: Configure gclient
+        run: |
+          depot_tools/gclient config https://github.com/google/shaka-packager.git --name=src --unmanaged
+
+      - name: Sync gclient
+        run: |
+          depot_tools/gclient sync
+
+      - name: Build
+        run: |
+          depot_tools/ninja -C src/out/${{ matrix.build_type }}${{ matrix.build_type_suffix }}
+
+      - name: Test
+        shell: bash
+        run: |
+          # NOTE: Some of these tests must be run from the "src" directory.
+          cd src
+          if [[ '${{ matrix.os_name }}' == 'osx' ]]; then
+            export DYLD_FALLBACK_LIBRARY_PATH=out/${{ matrix.build_type }}${{ matrix.build_type_suffix }}
+          fi
+          for i in out/${{ matrix.build_type }}${{ matrix.build_type_suffix }}/*test${{ matrix.exe_ext }}; do "$i" || exit 1; done
+          python out/${{ matrix.build_type }}${{ matrix.build_type_suffix }}/packager_test.py -v --libpackager_type=${{ matrix.lib_type }}_library
```

Comment:
You may want to add `echo` statements to show which thing is being run.  I don't think the tests print which executable it is coming from.

Reply:
Good call.  If this is all executed as one shell script, then `set -x` would do the trick nicely.